File: kafka-to-elastic-service/src/main/kotlin/com/chat/config/LoggingConfig.java
package com.chat.config;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

import java.lang.invoke.MethodHandles;

@Component
public class LoggingConfig {
    public final Logger service = getLogger("service");
    public final Logger performance = getLogger("performance");
    public final Logger kafkaToElastic = getLogger("kafkaToElastic");

    private static Logger getLogger() {
        return LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
    }

    private static Logger getLogger(String name) {
        return LoggerFactory.getLogger(name);
    }
}

File: asset-service/src/main/kotlin/com/chat/config/LoggingConfig.kt
package com.chat.config

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component
import java.lang.invoke.MethodHandles


fun getLogger(): Logger {
    return LoggerFactory.getLogger(MethodHandles.lookup().lookupClass())
}

fun getLogger(name: String): Logger {
    return LoggerFactory.getLogger(name)
}

@Component
class LoggingConfig {
    val s3 = getLogger("s3")
    val service = getLogger("service")
    val performance = getLogger("performance")
}